gapis/api/transform2/transform.go

```diff
@@ -45,8 +49,15 @@ type Transform interface {
 	// Resources are needed for the state mutation, therefore this should be called after mutation.
 	ClearTransformResources(ctx context.Context)
 
-	// RequiresAccurateState returns true if transform needs the observe the accurate state.
+	// RequiresAccurateState returns true if the transform needs to observe the accurate state.
 	RequiresAccurateState() bool
+
+	// RequiresInnerStateMutation returns true if the transform needs to mutate state during the transformation
+	RequiresInnerStateMutation() bool
+
+	// SetInnerStateMutationFunction sets a mutator function for making a branch in the transform
+	// to transform and mutate pending commands to be able to continue transform.
+	SetInnerStateMutationFunction(stateMutator StateMutator)
```

Comment:
With my C++ head on I'm not sure I like this being public. But I guess we have fewer choices in go.

Reply:
I am not very comfortable with Go but I am not sure if it is not public then can you implement this interface from another package because the transforms are under Vulkan package not transform.

Reply:
If we do as commented below and pass this in to TransformCommands as an extra argument, then the problem goes away. I think we should do that.